src/read/macho/dyld_cache.rs

```diff
@@ -28,7 +29,366 @@ where
     R: ReadRef<'data>,
 {
     data: R,
-    mappings: &'data [macho::DyldCacheMappingInfo<E>],
+    mappings: DyldCacheMappingSlice<'data, E, R>,
+}
+
+/// Information about a mapping.
+#[derive(Clone, Copy)]
+pub enum DyldCacheMapping<'data, E = Endianness, R = &'data [u8]>
+where
+    E: Endian,
+    R: ReadRef<'data>,
+{
+    /// Corresponds to struct dyld_cache_mapping_info from dyld_cache_format.h.
+    V1 {
+        /// The mapping endianness
+        endian: E,
+        /// The mapping data
+        data: R,
```

Comment:
I'm not keen on having `endian` and `data` in this..If you think it's simple to remove them then please do so, otherwise I can look into it more later.

Reply:
It didnt appear simple to me as I needed both data and endian in the most nested part of the chain iteration to get the file offsets